File: templates/golang/messaging.go
package main

import (
	"context"
	"log"
	"strconv"
	"sync"
	"time"

	amqp "github.com/rabbitmq/amqp091-go"
)

func HandleMessaging(wg *sync.WaitGroup) {
	defer wg.Done()
	conn, err := amqp.Dial("amqp://guest:guest@localhost:5672/")
	if err != nil {
		log.Panicf("Failed to connect to RabbitMQ: %s", err)
	}
	defer conn.Close()

	ch, err := conn.Channel()
	if err != nil {
		log.Panicf("Failed to open a channel: %s", err)
	}
	defer ch.Close()

	q, err := ch.QueueDeclare(
		"rpc_queue", // name
		false,       // durable
		false,       // delete when unused
		false,       // exclusive
		false,       // no-wait
		nil,         // arguments
	)
	if err != nil {
		log.Panicf("Failed to declare a queue: %s", err)
	}

	err = ch.Qos(
		1,     // prefetch count
		0,     // prefetch size
		false, // global
	)
	if err != nil {
		log.Panicf("Failed to set QoS: %s", err)
	}

	msgs, err := ch.Consume(
		q.Name, // queue
		"",     // consumer
		false,  // auto-ack
		false,  // exclusive
		false,  // no-local
		false,  // no-wait
		nil,    // args
	)
	if err != nil {
		log.Panicf("Failed to register a consumer: %s", err)
	}

	var forever chan struct{}

	go func() {
		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
		defer cancel()
		for d := range msgs {
			_, err := strconv.Atoi(string(d.Body))
			if err != nil {
				log.Panicf("Failed to convert body to integer: %s", err)
			}

			response := "" //{placeholder}""

			err = ch.PublishWithContext(ctx,
				"",        // exchange
				d.ReplyTo, // routing key
				false,     // mandatory
				false,     // immediate
				amqp.Publishing{
					ContentType:   "text/plain",
					CorrelationId: d.CorrelationId,
					Body:          []byte(response),
				})
			if err != nil {
				log.Panicf("Failed to publish a message: %s", err)
			}
			d.Ack(false)
		}
	}()

	log.Printf(" [*] Awaiting RPC requests")
	<-forever
}

File: templates/javascript/messaging.js
const amqp = require('amqplib/callback_api');
const { getEnvBool } = require('./utils');

function handleMessaging() {
    amqp.connect('amqp://localhost', function(error0, connection) {
        if (error0) {
            throw error0;
        }
        connection.createChannel(function(error1, channel) {
            if (error1) {
                throw error1;
            }
            const queue = 'rpc_queue';
            channel.assertQueue(queue, {
                durable: false
            });
            channel.prefetch(1);
            console.log(" [x] Awaiting RPC requests");

            channel.consume(queue, function reply(msg) {
                const requestData = JSON.parse(msg.content.toString());
                console.log("Received request:", requestData);

                const responseData = {{FUNCTION_CALL}} // Placeholder for response data

                channel.sendToQueue(msg.properties.replyTo,
                    Buffer.from(JSON.stringify(responseData)), {
                        correlationId: msg.properties.correlationId
                    });

                channel.ack(msg);
            });
        });
    });
}

module.exports = { handleMessaging };
